### PR TITLE
SCRUM-4646 add synonym loading for AGMs

### PR DESCRIPTION
**Pull request by cmpich**

### CODE REVIEW

File: src/main/java/org/alliancegenome/curation_api/model/ingest/dto/AffectedGenomicModelDTO.java [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 import org.alliancegenome.curation_api.constants.LinkMLSchemaConstants;
 import org.alliancegenome.curation_api.interfaces.AGRCurationSchemaVersion;
+import org.alliancegenome.curation_api.model.ingest.dto.slotAnnotions.NameSlotAnnotationDTO;
```
Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/model/ingest/dto/AffectedGenomicModelDTO.java:5]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.imports.UnusedImportsCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Unused import - org.alliancegenome.curation_api.model.ingest.dto.slotAnnotions.NameSlotAnnotationDTO.

File: src/main/java/org/alliancegenome/curation_api/model/entities/AffectedGenomicModel.java [hunk 2]
```diff
@@ -55,4 +52,13 @@ public class AffectedGenomicModel extends GenomicEntity {
 	@JsonView({ View.FieldsAndLists.class, View.GeneDetailView.class })
 	private List<ConstructGenomicEntityAssociation> constructGenomicEntityAssociations;
 
+	@IndexingDependency(reindexOnUpdate = ReindexOnUpdate.SHALLOW)
+	@FullTextField(analyzer = "autocompleteAnalyzer", searchAnalyzer = "autocompleteSearchAnalyzer")
+	@KeywordField(name = "synonyms_keyword", aggregable = Aggregable.YES, sortable = Sortable.YES, searchable = Searchable.YES, normalizer = "sortNormalizer")
+	@JsonView({ View.FieldsAndLists.class, View.AffectedGenomicModelView.class  })
```
Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/model/entities/AffectedGenomicModel.java:58]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.SingleSpaceSeparatorCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Use a single space to separate non-whitespace characters.

**Review by markquintontulloch — CHANGES_REQUESTED** [review]
Looks good but field needs adding to integration tests and test files

**Comment by markquintontulloch** [comment]
@cmpich The AGM bulk load tests and test files need updating too

**Review by markquintontulloch — APPROVED** [review]

**Pull request merged**